p2p/key.go

```diff
@@ -0,0 +1,122 @@
+package p2p
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+
+	crypto "github.com/tendermint/go-crypto"
+	cmn "github.com/tendermint/tmlibs/common"
+)
+
+type ID string
```

Comment:
@ebuchman what's the purpose of us defining `type ID string`?
If perhaps it's to add later utility methods like:
* `Validate() error`
* `Equals() bool`
then perhaps that's necessary.

Reply:
it's more so to clarify the allowed type when its being used eg. as a map index. We want to make sure we're not accidentally using the wrong string. If we use a type, the compiler enforces it :) ... `map[ID]whatever`

Reply:
Aye aye, gotcha. Thanks @ebuchman.